front/src/applications/stdcm/components/StdcmLoader.tsx

```diff
@@ -1,42 +1,104 @@
-import { forwardRef } from 'react';
+import { useEffect, useRef, type RefObject } from 'react';
 
 import { Button } from '@osrd-project/ui-core';
+import cx from 'classnames';
 import { useTranslation } from 'react-i18next';
 
-import stdcmLoaderImg from 'assets/pictures/views/stdcm_loader.jpg';
+import { LOADER_HEIGHT } from '../consts';
+import type { LoaderStatus } from '../types';
+
+const LOADER_BOTTOM_OFFSET = 32;
+const FORM_TOP_OFFSET = 120 + 68; // navbar + prev path heights
+const LAUNCH_BUTTON_HEIGHT = 40;
 
 type StdcmLoaderProps = {
   cancelStdcmRequest: () => void;
+  loaderStatus: LoaderStatus;
+  setLoaderStatus: (loaderStatus?: LoaderStatus) => void;
+  launchButtonRef: RefObject<HTMLDivElement>;
 };
 
-const StdcmLoader = forwardRef(
-  ({ cancelStdcmRequest }: StdcmLoaderProps, ref: React.Ref<HTMLDivElement>) => {
-    const { t } = useTranslation('stdcm');
-    return (
-      <div className="stdcm-loader-background">
-        <div
-          ref={ref}
-          className="stdcm-loader d-flex flex-column justify-content-center align-items-center"
-        >
-          <div className="stdcm-loader__wrapper">
-            <h1>{t('simulation.calculatingSimulation')}</h1>
-            <p>{t('simulation.averageRequestTime')}</p>
-          </div>
-          <div className="stdcm-loader__cancel-btn">
-            <Button
-              data-testid="cancel-simulation-button"
-              variant="Cancel"
-              label={t('simulation.stopCalculation')}
-              size="small"
-              onClick={cancelStdcmRequest}
-            />
-          </div>
-          <img src={stdcmLoaderImg} alt={t('simulation.pendingSimulation')} />
-          <p className="stdcm-loader__img-signature">{t('loaderImageLegend')}</p>
+const StdcmLoader = ({
+  cancelStdcmRequest,
+  loaderStatus,
+  setLoaderStatus,
+  launchButtonRef,
+}: StdcmLoaderProps) => {
+  const { t } = useTranslation('stdcm');
+  const loaderRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    // Depending on the scroll, change the position of the loader between fixed, sticky or absolute
+    const handleScroll = () => {
+      if (!loaderRef.current || !launchButtonRef.current) return;
+
+      const { scrollY, innerHeight } = window;
+
+      const isLoaderFitting =
+        innerHeight - launchButtonRef.current.getBoundingClientRect().top >
+        LOADER_HEIGHT + LOADER_BOTTOM_OFFSET;
+
+      if (launchButtonRef.current && isLoaderFitting) {
```

Comment:
You can handle only 3 cases:
```
  if the loader does not fit between the launch button and the bottom of the screen:
    setStatus('fixed-bottom');
    return;
  if the loader should stick to the top of the screen:
    setStatus('fixed-top');
  else:
    setStatus('absolute');
```

Reply:
Done

Reply:
I pushed another fixup to move the `LoadingStatus` state inside the `StdcmLoader` component. It seems to work the same and doesn't rerender all the form when we scroll.